WordPress/src/main/java/org/wordpress/android/ui/jetpackplugininstall/remoteplugin/JetpackRemoteInstallActivity.kt

```diff
@@ -0,0 +1,151 @@
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import dagger.hilt.android.AndroidEntryPoint
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.analytics.AnalyticsTracker.Stat.INSTALL_JETPACK_CANCELLED
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.login.LoginMode
+import org.wordpress.android.ui.ActivityLauncher
+import org.wordpress.android.ui.JetpackConnectionSource
+import org.wordpress.android.ui.JetpackConnectionUtils.trackWithSource
+import org.wordpress.android.ui.JetpackConnectionWebViewActivity
+import org.wordpress.android.ui.LocaleAwareActivity
+import org.wordpress.android.ui.RequestCodes
+import org.wordpress.android.ui.jetpackplugininstall.remoteplugin.JetpackRemoteInstallViewModel.JetpackResultActionData.Action.CONNECT
+import org.wordpress.android.ui.jetpackplugininstall.remoteplugin.JetpackRemoteInstallViewModel.JetpackResultActionData.Action.CONTACT_SUPPORT
+import org.wordpress.android.ui.jetpackplugininstall.remoteplugin.JetpackRemoteInstallViewModel.JetpackResultActionData.Action.LOGIN
+import org.wordpress.android.ui.jetpackplugininstall.remoteplugin.JetpackRemoteInstallViewModel.JetpackResultActionData.Action.MANUAL_INSTALL
+import org.wordpress.android.ui.accounts.HelpActivity
+import org.wordpress.android.ui.accounts.LoginActivity
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.jetpackplugininstall.install.compose.JetpackPluginInstallScreen
+import org.wordpress.android.ui.jetpackplugininstall.install.UiState
+import org.wordpress.android.util.extensions.setContent
+
+@AndroidEntryPoint
+class JetpackRemoteInstallActivity : LocaleAwareActivity() {
+    private val viewModel: JetpackRemoteInstallViewModel by viewModels()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            AppTheme {
+                val uiState by viewModel.liveViewState.observeAsState()
+                JetpackPluginInstallScreen(
+                    uiState = uiState ?: UiState.Initial(R.string.jetpack_plugin_install_initial_button),
+                    onDismissScreenClick = ::onBackPressed,
+                    onInitialButtonClick = viewModel::onInitialButtonClick,
+                    onDoneButtonClick = viewModel::onDoneButtonClick,
+                    onRetryButtonClick = viewModel::onRetryButtonClick,
+                    onContactSupportButtonClick = viewModel::onContactSupportButtonClick,
+                )
+            }
+        }
+        initViewModel(savedInstanceState)
+    }
+
+    override fun onBackPressed() {
+        if (viewModel.liveViewState.value?.showCloseButton == false) return
```

Comment:
💡 Would it make sense to move this `if` and the `trackWithSource` call to the ViewModel?

Reply:
I think it will not make a lot of difference in the code here as I still need to return some information saying the back should be blocked, but it's better in terms of testing the analytics. I will do the change.